server/src/main/java/org/elasticsearch/cluster/routing/allocation/ShardsAvailabilityHealthIndicatorService.java

```diff
@@ -147,6 +147,18 @@ public HealthIndicatorResult calculate(boolean explain) {
         DIAGNOSE_SHARDS_ACTION_GUIDE
     );
 
+    public static final String FIX_DELAYED_SHARDS_GUIDE = "http://ela.st/fix-delayed-shard-allocation";
+    public static final Diagnosis.Definition DIAGNOSIS_WAIT_FOR_OR_FIX_DELAYED_SHARDS = new Diagnosis.Definition(
+        "delayed_shard_allocations",
+        "Elasticsearch is not allocating some shards because they are marked for delayed allocation. Shards that are on nodes that leave "
+            + "the cluster are usually marked for delayed allocation because it is more efficient to see if the node returns to the "
+            + "cluster than to recover the shard immediately.",
+        "Restore the nodes that have left the cluster so that the shards can be restored. If this is not possible, Elasticsearch will "
+            + "allocate the shards when the delay has elapsed. If the shards must be made available immediately, configure the "
+            + "[index.unassigned.node_left.delayed_timeout] setting on the affected indices to 0 to force the allocation immediately.",
```

Comment:
Shall we recommend waiting for the configured `index.unassigned.node_left.delayed_timeout` timeout to lapse? (we can probably include the actual value of the configured timeout in the message)  - the allocation `AllocateUnassignedDecision` should contain this already

Cloud doesn't have the concept of nodes.

Reply:
> Cloud doesn't have the concept of nodes.

Good point, not sure how to describe the problem without nodes in this case, since this really only ever happens when nodes leave. Maybe the diagnosis message is just something like "The availability of these shards is temporarily disrupted, we are waiting to see if they come back"?

Agreed on just saying wait until the delayed timeout has elapsed. Small snag on displaying the timeout though is if multiple indices have different values in them. The message could get quite verbose if we list each of them. Maybe something worth putting into the details?

Reply:
>"The availability of these shards is temporarily disrupted, we are waiting to see if they come back"

That sounds good. Should we mention nodes as a second sentence? I feel like the users who are aware of nodes would appreciate that? 

> Maybe something worth putting into the details?

Good shout about the possible complexity. I'd say the details is a good section for this.

Reply:
I just removed all mention of nodes. The message is a little simpler now. I also decided against adding the delay to the details section. I doubt many people are setting this property to wildly different values, and reporting that it is 1 minute by default is probably not worth the effort when we can touch on it in the troubleshooting guide.